### PR TITLE
(PDB-2097) Disable Puppet Server startup checkins to Dujour

### PR DESCRIPTION
**Pull request by ajroetker**
Prior to this commit Puppet Server would checkin to Dujour when
puppetserver was started up because we munged `/etc/hosts` after
puppetserver had already been installed. This commit separates the
installation of puppet and puppetserver so that we can put the munging
of `/etc/hosts` in the middle.

This commit also removes a lot of dead code that was getting in the way
of the refactor.

### CODE REVIEW

**Comment by ajroetker** [comment]
@puppetlabs-jenkins test this please

File: acceptance/helper.rb [hunk 1]
```diff
@@ -744,123 +726,43 @@ def object_diff(obj1, obj2)
...
-      commands = ["cd #{target}",
-                  "remote rm origin",
-                  "remote add origin #{repo}",
-                  "fetch origin #{opts[:refspec]}",
-                  "clean -fdx",
-                  "checkout -f #{rev}"]
-      on host, commands.join(" && git ")
+  def munge_etc_hosts_to_disable_dujour_checkins
+    databases.each do |database|
+      stub_hosts_on(database, 'updates.puppetlabs.com' => '127.0.0.1')
```
Comment (ajroetker) [inline acceptance/helper.rb]: @nwolfe Here's the example of the `stub_hosts_on` vs. `apply_manifest_on` for dealing with the acceptance test pre_suite business for dujour.

**Comment by ajroetker** [comment]
@puppetlabs-jenkins test this please

**Comment by kbarber** [comment]
Doesn't this problem also apply to stable?

**Comment by mullr** [comment]
@ajroetker should we close this PR now and get the code from a roll-up merge?

**Comment by ajroetker** [comment]
@mullr This code change is more comprehensive, I've already rebased on a roll-up locally, just waiting for master to be green.

**Comment by ajroetker** [comment]
I'm gonna close this PR and make a ticket to do a significant removal of the dead code in the acceptance tests.

**Pull request closed without merging**